### PR TITLE
makes BehaviorSubject emit an Error, if the latest event was an Error…

### PR DESCRIPTION
**Pull request by frankpepermans**
…, https://github.com/ReactiveX/rxdart/issues/227

### CODE REVIEW

File: test/subject/behavior_subject_test.dart [hunk 1]
```diff
@@ -35,6 +35,20 @@ void main() {
       await expectLater(subject, emits(3));
     });
 
+    test('emits errors to every subscriber', () async {
```
Comment (brianegan) [inline test/subject/behavior_subject_test.dart]: Maybe add a check in here for `subject.value, isNull` as well and the opposite test for the `addError` -> `add` condition?

Comment (frankpepermans) [reply test/subject/behavior_subject_test.dart]: sure! added meanwhile

**Comment by frankpepermans** [comment]
@brianegan OK to merge for ya?

**Comment by brianegan** [comment]
Yes sir!

**Pull request merged**